chef/cookbooks/hyperv/recipes/openstack_install_networking-hyperv.rb

```diff
@@ -0,0 +1,26 @@
+raise if not node[:platform_family] == "windows"
+
+cookbook_file "#{node[:cache_location]}#{node[:openstack][:networking_hyperv][:file]}" do
+	source node[:openstack][:networking_hyperv][:file]
+	not_if { ::File.exist?(node[:openstack][:networking_hyperv][:file]) }
+end
+
+windows_batch "unzip_networking-hyperv" do
+  code <<-EOH
+  #{node[:sevenzip][:command]} x #{node[:cache_location]}#{node[:openstack][:networking_hyperv][:file]} -o#{node[:openstack][:location]} -r -y
+  #{node[:sevenzip][:command]} x #{node[:openstack][:location]}\\dist\\#{node[:openstack][:networking_hyperv][:name]}-#{node[:openstack][:networking_hyperv][:version]}.tar -o#{node[:openstack][:location]} -r -y
+  rmdir /S /Q #{node[:openstack][:location]}\\dist
+  ren #{node[:openstack][:location]}\\#{node[:openstack][:networking_hyperv][:name]}-#{node[:openstack][:networking_hyperv][:version]} #{node[:openstack][:networking_hyperv][:name]}
+  EOH
+  not_if { ::File.exists?("#{node[:openstack][:location]}\\#{node[:openstack][:networking_hyperv][:name]}") }
```

Comment:
Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.<br>Metrics/LineLength: Line is too long. [109/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)